numpy/core/arrayprint.py

```diff
@@ -521,7 +513,10 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if a.shape == () and a.dtype.names is None:
```

Comment:
I though I understood you suggestion but now I'm not sure. How does it change the behavior of this PR?

Also, note that `array_str` in `numeric.py` calls `array2string(..., style=str)`, while `array_repr` calls `array2string(...)` (ie, default value of style). That is the main use/effect of the style argument in numpy.